precompile/allow_list.go

```diff
@@ -70,6 +72,28 @@ func (c *AllowListConfig) Equal(other *AllowListConfig) bool {
 	return true
 }
 
+// Verify returns an error if there is an overlapping address between admins and enableds
+func (c *AllowListConfig) Verify() error {
+	// check if both lists are empty
+	if len(c.EnabledAddresses) != 0 && len(c.AllowListAdmins) != 0 {
+		enabledMap := make(map[common.Address]struct{})
+		for _, enabledAddr := range c.EnabledAddresses {
+			if _, ok := enabledMap[enabledAddr]; !ok {
+				enabledMap[enabledAddr] = struct{}{}
+			}
+		}
+		for _, adminAddr := range c.AllowListAdmins {
+			if _, ok := enabledMap[adminAddr]; ok {
+				return fmt.Errorf("cannot set address %s as both admin and enabled", adminAddr)
+			}
+		}
+	}
+	return nil
+}
+
+// Enum constants for valid AllowListRole
+type AllowListRole common.Hash
+
```

Comment:
Can we move all of the AllowList type, functions and the declarations of the valid ones into one file?

I preferred the type declaration right above the declaration of the enum values, but I think it's even cleaner if we move it to a separate file

Reply:
You mean moving `AllowListRole` declaration and funcs into a new file? Or you mean moving all of AllowList types (like ContractDeployer/TxAllowList) into one file?

Reply:
Yup I meant moving `AllowListRole` declaration and functions into a new file